### PR TITLE
Use Device specific gesture configuration for scroll views

### LINKED ISSUE #87322: Flutter should use the platform configured touch slop value for Android
Currently the framework uses a constant `kTouchSlop = 18` (logical pixels) for all touch interactions. This has historically been the source of some controversy.

The Android Framework provides a touch slop value via the [`ViewConfiguration`](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/ViewConfiguration.java;l=718) class. This value is in physical pixels (slight guess, though logical pixels are usually referred to as dp/dips).

On my personal Pixel 4 device, querying the view configuration with the following snippet:

```java
Log.i("Flutter", "value= "  + ViewConfiguration.of(getApplicationContext).getScaledTouchSlop());
```

I get a value of `22`. Converting from physical to logical pixels with a dpr of `2.75` gives `8` physicals pixels - less than half the touch slop we currently use.

I believe this discrepancy contributes to some amount of the reported lag, since a flutter list view will feel less responsive than a native list view on the same phone because the scrolling drag gesture will be accepted slightly later on Flutter. At the same time, there may be cases where manufacturers have configured this value higher due to device specific touch input performance.

Reading this value and forwarding to the framework could be done in a similar method to all of the existing view configuration/media query logic. Drag gesture recognizers could be configured to use this value and fall back to the existing constant.

Other platforms may have similar features that we should keep an eye out for.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/87322

Wires up the window specific GestureSettings into a dpr transformed DeviceGestureSettings. This can be configured on the base GestureDetector, and then each gesture detector can choose which (if any) configuration it wants to use.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I don't really want to hit everything at once, but making this optional for the sake of backwards compat - maybe we can migrate to this over time. @goderbauer for thoughts?

File: packages/flutter/lib/src/gestures/force_press.dart [hunk 1]
```diff
@@ -249,7 +249,7 @@ class ForcePressGestureRecognizer extends OneSequenceGestureRecognizer {
         if (pressure > startPressure) {
           _state = _ForceState.started;
           resolve(GestureDisposition.accepted);
-        } else if (event.delta.distanceSquared > computeHitSlop(event.kind)) {
+        } else if (event.delta.distanceSquared > computeHitSlop(event.kind, null)) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/force_press.dart:252]: Why null over gestureSettings.touchSlop?

File: packages/flutter/lib/src/gestures/multidrag.dart [hunk 2]
```diff
@@ -497,17 +504,17 @@ class VerticalMultiDragGestureRecognizer extends MultiDragGestureRecognizer {
 
   @override
   MultiDragPointerState createNewPointerState(PointerDownEvent event) {
-    return _VerticalPointerState(event.position, event.kind);
+    return _VerticalPointerState(event.position, event.kind, null);
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/multidrag.dart:507]: why null?

File: packages/flutter/lib/src/gestures/events.dart [hunk 3]
```diff
@@ -2099,28 +2099,28 @@ class PointerCancelEvent extends PointerEvent with _PointerEventDescription, _Co
 }
 
 /// Determine the appropriate hit slop pixels based on the [kind] of pointer.
-double computeHitSlop(PointerDeviceKind kind) {
+double computeHitSlop(PointerDeviceKind kind, double? deviceTouchSlop) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:2102]: should these just take a DeviceGestureSettings obj and the method can pick the appropriate value from it?

File: packages/flutter/lib/src/gestures/multitap.dart [hunk 4]
```diff
@@ -244,6 +246,7 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
       event: event,
       entry: GestureBinding.instance!.gestureArena.add(event.pointer, this),
       doubleTapMinTime: kDoubleTapMinTime,
+      touchSlop: null,
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/multitap.dart:249]: Why null?

File: packages/flutter/lib/src/gestures/multitap.dart [hunk 5]
```diff
@@ -879,6 +885,7 @@ class SerialTapGestureRecognizer extends GestureRecognizer {
       invokeCallback<void>('onSerialTapDown', () => onSerialTapDown!(details));
     }
     final _TapTracker tracker = _TapTracker(
+      touchSlop: null,
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/multitap.dart:888]: Why null?

**Review by goderbauer — COMMENTED** [review]
In every place where null is passed over the real device value a comment would be good explaining why we want the framework fallback rather than the device-provided value.

File: packages/flutter/lib/src/gestures/gesture_settings.dart [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    return DeviceGestureSettings(
+      touchSlop: physicalTouchSlop == null ? null : physicalTouchSlop / window.devicePixelRatio
+    );
+  }
+
+  /// The touch slop value in logical pixels, or `null` if it was not set.
+  final double? touchSlop;
+
+  /// The touch slop value for pan gestures, in logical pixels, or `null` if it
+  /// was not set.
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/gesture_settings.dart:36]: Looks like there is no way to actually set it independently? Maybe document that it's either null or 2* touchSlop if touchSlop is set?

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 7]
```diff
@@ -104,6 +105,7 @@ class MediaQueryData {
     this.disableAnimations = false,
     this.boldText = false,
     this.navigationMode = NavigationMode.traditional,
+    this.gestureSettings = const DeviceGestureSettings(touchSlop: 16.0)
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:108]: Can this use kTouchSlop instead of hard-coding 16?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/gestures/gesture_settings.dart (hunk 6 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/gestures/gesture_settings.dart:36]: Will do. In the future we could support setting it separately if need be.

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 7 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/widgets/media_query.dart:108]: Done

File: packages/flutter/lib/src/gestures/events.dart [hunk 8]
```diff
@@ -2099,28 +2099,28 @@ class PointerCancelEvent extends PointerEvent with _PointerEventDescription, _Co
...
     case PointerDeviceKind.mouse:
       return kPrecisePointerPanSlop;
     case PointerDeviceKind.stylus:
     case PointerDeviceKind.invertedStylus:
     case PointerDeviceKind.unknown:
     case PointerDeviceKind.touch:
-      return kPanSlop;
+      return deviceTouchSlop != null ? (2 * deviceTouchSlop) : kPanSlop;
   }
 }
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/gestures/events.dart:2127]: I'm not quite sure what to do with the scale slop right now.

**Pull request merged**